internal/pkg/api/server/buffer/dependencies/store.go

```diff
@@ -65,3 +72,62 @@ func (c *ConfigStore) CreateReceiver(ctx context.Context, receiver model.Receive
 
 	return nil
 }
+
+// GetReceiver fetches a receiver from the store.
+//
+// This method returns nil if no receiver was found.
+func (c *ConfigStore) GetReceiver(ctx context.Context, projectId int, receiverId string) (r *model.Receiver, err error) {
+	logger, tracer, client := c.logger, c.tracer, c.etcdClient
+
+	_, span := tracer.Start(ctx, "kac.api.server.buffer.dependencies.store.GetReceiver")
+	defer telemetryUtils.EndSpan(span, &err)
+
+	key := ReceiverKey(projectId, receiverId)
+
+	logger.Debugf(`GET "%s"`, key)
+	resp, err := client.KV.Get(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+
+	// No receiver found
+	if len(resp.Kvs) == 0 {
+		logger.Debugf(`No receiver "%s" found`, key)
+		return nil, nil
+	}
+
+	logger.Debugf(`Decoding "%s"`, key)
+	receiver := &model.Receiver{}
+	if err = json.DecodeString(string(resp.Kvs[0].Value), receiver); err != nil {
+		return nil, err
+	}
+
+	return receiver, nil
+}
+
+func (c *ConfigStore) ListReceivers(ctx context.Context, projectId int) (r []*model.Receiver, err error) {
+	logger, tracer, client := c.logger, c.tracer, c.etcdClient
+
+	_, span := tracer.Start(ctx, "kac.api.server.buffer.dependencies.store.ListReceivers")
+	defer telemetryUtils.EndSpan(span, &err)
+
+	key := ProjectKey(projectId)
+
+	logger.Debugf(`GET "%s"`, key)
+	resp, err := client.KV.Get(ctx, key, etcd.WithPrefix())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Nevrátí to ale jen první stránku? (Zrovna na to koukám, protože potřebuju listovat exporty. Podle tohoto mi přijde, že bude potřeba při listování čehokoli ohandlovat stránkování: https://www.compose.com/articles/utilizing-etcd3-with-go/#pagination

Reply:
Tady je `WithLimit`, kde popisujou že `limit: 0` znamená `no limit`. https://github.com/etcd-io/etcd/blob/2a0e92fac0126b5fb2c629cb343955cff960ca6f/client/v3/op.go#L341

Tady je `OpGet`, který se volá při `KV.Get` a nemá tam nastavený `limit`, takže bude `0`: https://github.com/etcd-io/etcd/blob/2a0e92fac0126b5fb2c629cb343955cff960ca6f/client/v3/op.go#L227

Tím pádem to nemusíme řešit

Reply:
To jako že by vraceli najednou úplně všechny výsledky do paměti? Co se pak stane když to přestřelí třeba ten jejich request limit? https://etcd.io/docs/v3.3/dev-guide/limit/

Reply:
Ja si tiez myslim, ze strankovat to treba iba ked sa pouzije `WithLimit`, ale neskusal som to.

Reply:
Mame v design doc limits na pocet receivers a exports per project.
V @jprochazk , ktory to vracia per project je to teda OK.
V @JakubMatejka usecase, kde bude nacitavat asi vsetko, zo vsetkych projektov to nemusi byt OK.

Reply:
(len nesmieme tie limity zabudnut implementovat, pri vytvarani daneho resource)

Reply:
Tím pádem tady ještě musím přidat check na receiver count

Reply:
Nó, já jsem teda zvyklej u každý databáze řešit jako první věc u listování stránkování a přijde mi zvláštní že by se zrovna v etcd s ní defaultně nepočítalo (např. i Dynamo DB defaultně stránkuje), ale když myslíte. 🙂

Reply:
No podla docs je to tak, ze ked limit nie je tak sa vrati vsetko, ... kym nedojde nejaky resource. Je to na programatorovi aby si to dobre premyslel :slightly_smiling_face: 

Ked vieme ze tam nebudeme mat viac ako `100` zaznamov per project, tak nevidim problem v tom, to pouzit bez `WithLimit`.

Aj tak tu response generujeme v memory, a teda vsetko bude v memory, ... a nemame dovod prerabat vystup na streaming (goa to podporuje).

Ale kludne sa moze spravit nejaky `type Pager interface { Next() <Page>, bool }`, ktory bude mat default limit `100` a budeme ho pouzivat vsade. Aby sme to mali este viac `safe`. 

Alebo tam dat `WithLimit(100)` a spracovat iba prvu stranku, kedze viac zaznamov nepovolujeme. To by bolo `safe` aj z hladiska apky.

Reply:
Oni s pagingem počítají, ale by default to nedělá paging. Koukni se na ty linky, co jsem poslal o trochu výš. Jestli to je potřeba pro import, tak by to měla být zvlášť metoda na `ConfigStore` (`ListReceiversPaged` nebo podobně)

Reply:
Jo, ta poslední možnost mi přijde rozumná. Určitě bych se tiše nespolíhal na to že nereachneme limity. Za nějakou dobu přijde z Produktu požadavek na navýšení/zrušení toho našeho omezení 100 záznamů na projekt, zapomene se na to a bude zaděláno na pěkný mnhohahodinový debug proč to klientovi nefunguje. 🙂

Reply:
Mně to nepřijde jako správný přístup. S debugem ti to moc nepomůže, protože pak místo toho, aby to třeba spadlo na out of memory (jestli vůbec), tak to bude blbnout klientovi, že mu nějaký receivery náhodně nefungujou.

Taky ten `CreateReceiver` není atomický (nemá zámky), takže se může stát, že dvě instances API vytvoří receivery v případě, když už je jich tam 99, takže jich bude 101. Pak se zase bude dít to, že taky náhodně budeme "ignorovat" nějaké receivery.

Reply:
Lepší by imho bylo řešit paging až budeme potřebovat paging, což bude až když dojde produkt s tím, že chcou zvýšit limit pro některý projekty. Nepřijde mi to jako technický dluh, jen to prostě nebylo specifikované, tak si tu práci můžeme něchat na později.

Reply:
Jj presne ako pises (zamok sa tam da pridat, ak chceme zabezpecit presne `100`), ale pride mi v pohode ze jednemu klientovi, ktory prekrocil limit nieco nezobrazime (jedna sa o konfiguracne api, nie import).

Je to ovela lepsie ako zhodit cele API pre out-of-memory.
Etcd ti vrati (myslim) ci nasleduje dalsia stranka, mozes zalogovat chybu ....

Reply:
A mas v pohode aj tyzden na to, aby si to opravil, ... ale ked ti nepojde cele API, tak to budes do noci debugovat a opravovat.

Reply:
Rozdíl je ale v tom, že OOM vyskočí u nás hned, na ty náhodně nefungující receivery budeme čekat až přijde ticket od klienta

Reply:
Nebudeme, ako som pisal, zaloguj chybu a vieme o tom hned.
`Unexpected state: found more than 100 receivers.`

Reply:
Ale zaroven bude vsetko fungovat (az na jedneho klienta, ktory uvidi `100` a nie `102` receivers).

Reply:
(inak super diskusia, ... myslim, ze spolu dojdeme k lepsiemu rieseniu, ako by prisiel kazdy z nas samostatne)

Reply:
Jediný problém, který teď může nastat je to, že někdo bude mít 102 receiverů místo 100. Což ale imho problém není.

Reply:
A ak by si iba pridal `WithLimit()` a nic viac nerobil? Bol by to problem?
Zachrani ta to, ked bude niekde inde v kode chyba a omylom nagenerujes do `etcd` milion receivers per project.
+ jeden if `result.more -> log.Error`.

Reply:
Když přidám jen `WithLimit` a nic víc, tak tam bude problém, že někomu náhodně nebudou fungovat některý receivery, protože je budeme ignorovat.

Reply:
A to zalogujeme a vyriesime, ale ak sa stane vacsi problem tak nas to ochrani. 

Navrhujem, dat tam teraz `WithLimit`.
A ked budes riesit zamok ohladom `Update`, tak ho skopirujes aj tu, a budeme vsetci spokojny :slightly_smiling_face:

Reply:
Zámky to všechno hodně zkomplikují, protože budou potřeba na více místech:
- Create receiver
- Delete receiver

+ ještě by musely být zámky na exports:
- Create export
- Delete export 

Proto mi přijde lepší to zatím takhle nechat a řešit paging až budeme definitivně vědět, že budeme chtít u některých projektů vyšší limity.

Reply:
Je to práce navíc pro kterou nemáme use-case.

Reply:
> Zámky to všechno hodně zkomplikují, protože budou potřeba na více místech:

To budu 2-3 riadky na tychto miestach, ale asi sa o tom dohadujeme uz dlhsie ako by sa to programovalo.
Tak ak mas s tym problem, tak zaloz task do rovnakeho epicu, a prehodnotime to nakoniec.

Reply:
| Je to práce navíc pro kterou nemáme use-case.

No nejake use-cases sme nasli, proste sa moze spravit chyba inde.

Reply:
https://keboola.atlassian.net/browse/BA-84

Reply:
super, este si chcem PR vyskusat, dostanem sa k tomu najneskor vecer